build_helpers/publishPackage.sh

```diff
@@ -1,35 +1,70 @@
 #!/bin/bash
+# example usages:
+# ./publishPackage.sh --branch=v1.0.x
+# ./publishPackage.sh --branch=v1.1.x --latest
+# ./publishPackage.sh --branch=v1.0-beta --beta
+
 set -e
 
 # Defaults that can be overridden in the CLI (except the cookie file)
-export BETA=0
+export BETA=false
+export LATEST=false
+export BRANCH=""
 
+# Parse flags
 while (( "$#" )); do
   opt=$1
   VALUE=$2
   # Make sure to shift here and any option that uses the $VALUE
   shift
   case $opt in
-    --beta) BETA=$VALUE; shift ;;
+    --beta)
+      BETA=true
+      ;;
+
+    --latest)
+      LATEST=true
+      ;;
+
+    --branch)
+      if [ -z $VALUE ] ; then
+        echo "Please specify branch name"
+        exit 192
+      fi
+      BRANCH=$VALUE;
+      shift
+      ;;
+
     * ) # Invalid flag
       echo "Invalid flag: $opt"
       exit 192
       ;;
   esac
 done
 
-current_version=$(node -p "require('./package').version")
+# branch name is required
```

Comment:
Earlier we always pushed under master. 
Now that we might run the script from multiple branches like v1.0.x or v.1.1.x, let's have the user specify the upstream branch.